### PR TITLE
[BugFix] Fix wrong state of 'is_result_sink' of query context

### PR DESCRIPTION
**Pull request by liuyehcf**
Introduced by #29901

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/exec/pipeline/query_context.cpp [hunk 1]
```diff
@@ -197,7 +197,7 @@ std::shared_ptr<QueryStatistics> QueryContext::intermediate_query_statistic() {
 }
 
 std::shared_ptr<QueryStatistics> QueryContext::final_query_statistic() {
-    DCHECK(_is_result_sink) << "must be the result sink";
+    DCHECK(_is_final_sink) << "must be the result sink";
```
Comment (kangkaisen) [inline be/src/exec/pipeline/query_context.cpp:200]: error message

Comment (liuyehcf) [reply be/src/exec/pipeline/query_context.cpp:200]: updated

**Review by trueeyu — DISMISSED** [review]

File: be/src/exec/pipeline/query_context.cpp [hunk 2]
```diff
@@ -176,7 +176,7 @@ std::shared_ptr<QueryStatisticsRecvr> QueryContext::maintained_query_recv() {
 std::shared_ptr<QueryStatistics> QueryContext::intermediate_query_statistic() {
     auto query_statistic = std::make_shared<QueryStatistics>();
     // Not transmit delta if it's the result sink node
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply be/src/exec/pipeline/query_context.cpp]: updated

**Review by trueeyu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.5

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-3.0

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**